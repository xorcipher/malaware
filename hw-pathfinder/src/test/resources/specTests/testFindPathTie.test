# Find a path with Dijkstra's algorithm when
# the nodes entered have two paths between them
# with the same cost. This should result
# in a path with the least number of hops.

CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B
AddNode simpleGraph C

AddEdge simpleGraph A C 5.0
AddEdge simpleGraph A B 2.5
AddEdge simpleGraph B C 2.5

FindPath simpleGraph A C