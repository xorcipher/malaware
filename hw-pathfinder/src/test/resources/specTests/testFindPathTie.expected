# Find a path with Dijkstra's algorithm when
# the nodes entered have two paths between them
# with the same cost. This should result
# in a path with the least number of hops.

created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph
added node C to simpleGraph

added edge 5.000 from A to C in simpleGraph
added edge 2.500 from A to B in simpleGraph
added edge 2.500 from B to C in simpleGraph

path from A to C:
A to C with weight 5.000
total cost: 5.000